framework/db/conditions/BetweenColumnsCondition.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace yii\db\conditions;
+
+use yii\base\InvalidArgumentException;
+use yii\db\ExpressionInterface;
+use yii\db\Query;
+
+/**
+ * Class BetweenColumnCondition represents a `BETWEEN` condition where
+ * values is between two columns. For example:
+ *
+ * ```php
+ * new BetweenColumnsCondition(42, 'BETWEEN', 'min_value', 'max_value')
+ * // Will be build to:
+ * // 42 BETWEEN min_value AND max_value
+ * ```
+ *
+ * And a more complex example:
+ *
+ * ```php
+ * new BetweenColumnsCondition(
+ *    new Expression('NOW()'),
+ *    'NOT BETWEEN',
+ *    (new Query)->select('time')->from('log')->orderBy('id ASC')->limit(1),
+ *    'update_time'
+ * );
+ *
+ * // Will be built to:
+ * // NOW() BETWEEN (SELECT time FROM log ORDER BY id ASC LIMIT 1) AND update_time
+ * ```
+ *
+ * @author Dmytro Naumenko <d.naumenko.a@gmail.com>
+ * @since 2.0.14
+ */
+class BetweenColumnsCondition implements ConditionInterface
+{
+    /**
+     * @var string $operator the operator to use (e.g. `BETWEEN` or `NOT BETWEEN`)
+     */
+    protected $operator;
```

Comment:
Do we need these to be protected instead of private?

Reply:
Well, to be honest all properties is conditions may be private. They are not intended to be reused, except abstract class `ConjunctionCondition`. I can make all 'em private with a separate PR

Reply:
Why separate PR? Let's make em private, I think.

Reply:
Ok